### DISCUSSION TITLE
Support Google Keep style Archiving of Notes

### DISCUSSION THREAD

**Discussion by vHanda** [GitJournal/GitJournal#712] [Feature Requests]
A user has requested we support Google Keep style archiving of Notes. One possible way of supporting this would be for monitoring the yaml metadata `archived`, and those notes do not show up normally, and we have a separate 'Archived' section in the sidebar.

Another possibility would just be to move them to Archived folder. Maybe that would be better? I'd love some feedback.

**Comment by AudunVN** [comment]
I thought a bit about exactly this after moving over from Google Keep using your export tool yesterday, since I had around 300-400 archived notes I wanted to move with me to GitJournal, but still keep out of my "active" notes. 

I personally decided moving my archived notes to a separate folder was a better solution. It's portable and keeps archived notes from cluttering up my folders in anything that doesn't support the `archived` YAML frontmatter value, despite some initial cost due to having to move all my old notes marked `archived`  to a separate folder.

User preference will likely depend a lot on how they prefer to interact with their notes. If one doesn't need to use the repository much outside GitJournal then the difference in user experience between using `archived` and a dedicated folder for archived notes could be fairly small. In use cases like mine, however, where one uses external Markdown (or even plain text) viewers/editors and file managers to interact with notes, folders seem to have a clear advantage.

PS: Thank you very much for your excellent work on GitJournal, very glad to have discovered and moved over to it from Google Keep! Was fairly straightforward to do thanks to your export tool 😊

**Comment by ModProg** [comment]
I do think folder based is better, the metadata should hold as little information as possible in my opinion, to stay as compatible with plain Markdown as possible.

The question remains tho, if the archive folder should contain the whole folder structure or if the hierarchy should be flattened? 
Probably not necessary as most users wont have 20 folder deep structures for their notes.

**Comment by mikegr** [comment]
I miss this feature. I have organized my notes in folders and I would be fine to have a `.archived` sub folder in each folder.

**Comment by radupotop** [comment]
> Thank you for filling this. I'm closing it as it's a duplicate of https://github.com/GitJournal/GitJournal/issues/387 . Would you mind voting on that issue please? Also I'm not sure what mechanism there should be for archiving -
> 
> 1. Add `archived: true` in the YAML MetaData - This doesn't work for markdown files without yaml or txt or other kinds of files.
> 2. Add a `.archived` folder in the directory where the note exists
> 3. Add a top level 'Archived' folder and mirror the directory structure - This seems difficult to do as at some point the directory structure is going to go out of sync.
> 
> I'm leaning towards (2), as that is the simplest. Also, folders starting with a `.` are already ignored by GitJournal so this will be much easier to implement.
> 

Thanks for taking the time to reply.  

I think option 1 would work very well.

As for option 3 (which I proposed initially)  - I didn't realise you would need to duplicate the original directory structure under the Archive folder. 

I guess you could just dump files in that folder without any structure.  
But then you could run into the issue of having duplicate names. 
You could solve this by renaming all archived notes to UUIDs.

**Comment by timurguseynov** [comment]
@vHanda is the IOS version has this feature?
